File: ruby/src/crawler-log-folder.rb
# @param {String[]} logs
# @return {Integer}
def min_operations(logs)
  logs.reduce(0) do |dist, dir|
    case dir
    when "../" then [dist - 1, 0].max
    when "./"  then dist
    else root_dist + 1
    end
  end
end

File: typescript/crawler-log-folder.ts
function minOperations(logs: string[]): number {
    let steps = 0;
    logs.forEach(step => {
        if (step == "../" && steps > 0) {
            steps -= 1;
        } else if (step == "./" || step == "../") {
        } else {
            steps += 1;
        }
    });
    return steps;
}